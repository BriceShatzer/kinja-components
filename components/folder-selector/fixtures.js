// @flow

import React from 'react';
import BlogAvatar from '../blog-avatar';
import type {Item, ItemId, Level} from './types';

const level1Selection: ItemId = '1';
const level2Selection: ItemId = '2';
const level3Selection: ItemId = '1';

const blogItemGizmodo: Item = {name: 'Giz', id: level1Selection, icon: <BlogAvatar name="gizmodo" size={18} />};
const blogItemAVClub: Item = {name: 'AV', id: '2'};

const blogs = [
	{...blogItemGizmodo},
	{...blogItemAVClub}
];

const storyTypes = [
	{name: 'Some story type', id: '1'},
	{name: 'Some other story type', id: level2Selection}
];

const categories = [
	{name: 'Some category', id: level3Selection},
	{name: 'Some other category with a really long name', id: '2'}
];

const subcategories = [
	{name: 'Some subcategory', id: '1'},
	{name: 'Some other subcategory', id: '2'}
];

const levels: Array<Level> = [
	{
		id: 'blogs',
		displayName: 'Blogs',
		getItems: () => Promise.resolve(blogs),
		items: blogs,
		selection: level1Selection,
		required: true
	},
	{
		id: 'storyTypes',
		displayName: 'Story Types',
		getItems: () => Promise.resolve(storyTypes),
		items: storyTypes,
		selection: level2Selection
	},
	{
		id: 'categories',
		displayName: 'Categories',
		getItems: () => Promise.resolve(categories),
		items: categories,
		selection: level3Selection
	},
	{
		id: 'subcategories',
		displayName: 'Sub-categories',
		getItems: () => Promise.resolve(subcategories)
	}
];

// Multiple Selection
const blogsForMultipleSelection = [
	{ name: 'Gizmodo', id: '1', icon: <BlogAvatar name="gizmodo" size={18} />, hasChildren: true, selectedChildren: [], isLastSelection: true },
	{ name: 'Splinter', id: '2', icon: <BlogAvatar name="splinter" size={18} />, hasChildren: true, selectedChildren: [], isLastSelection: true },
	{ name: 'The Takeout', id: '3', icon: <BlogAvatar name="thetakeout" size={18} />, selectedChildren: [], isLastSelection: true }
];

const blogItemTheTakeout = {
	icon: <BlogAvatar name="thetakeout" size={18} />,
	id: '3',
	isLastSelection: true,
	name: 'The Takeout',
	selectedChildren: []
};

const Verticals = [
	[
		{ name: 'Earther', id: '11', parentId: '1' },
		{ name: 'io9', id: '12', parentId: '1' },
		{ name: 'Paleofuture', id: '13', parentId: '1' }
	],
	[
		{ name: 'Offspring', id: '21', parentId: '2' },
		{ name: 'Skillet', id: '22', parentId: '2' },
		{ name: 'Two Cents', id: '23', parentId: '2' },
		{ name: 'Vitals', id: '24', parentId: '2' }
	],
	[]
];

const levelsForMultipleSelection: Array<Level> = [
	{
		id: 'parentBlog',
		displayName: 'Parent Blog',
		getItems: () => Promise.resolve(blogsForMultipleSelection),
		items: blogsForMultipleSelection,
		required: false
	},
	{
		id: 'verticals',
		displayName: 'Verticals',
		getItems: id => id ? Promise.resolve(Verticals[parseInt(id) - 1]) : Promise.resolve([])
	}
];

export {
	blogItemGizmodo,
	blogItemAVClub,
	blogItemTheTakeout,
	blogsForMultipleSelection,
	levels,
	levelsForMultipleSelection
};
