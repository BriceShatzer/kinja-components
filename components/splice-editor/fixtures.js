/* @flow */

import React from 'react';
import Post from 'kinja-magma/models/Post';
import Blog from 'kinja-magma/models/Blog';
import BlogAvatar from '../blog-avatar';

import post from 'kinja-components/__stubs__/stubbedPostResponse.json';
import blog from 'kinja-components/__stubs__/gizmodo.json';


export const currentPost = Post.fromJSON({
	...post,
	...{
		headline: post.headline,
		properties: JSON.stringify({
			...JSON.parse(post.properties),
			...{
				isEmbiggened: true
			}
		})
	}
});

export const currentBlog = Blog.fromJSON({
	...blog,
	...{
		properties: {
			...blog.properties,
			...{
				blogGroup: 'gizmodo'
			}
		}
	}
});


export const selectedBlogs = [
	{
		icon: <BlogAvatar name="gizmodo" size={18} />,
		id: 12,
		name: 'io9',
		parentId: '1'
	},
	{
		icon: <BlogAvatar name="gizmodo" size={18} />,
		id: 13,
		name: 'Paleofuture',
		parentId: '1'
	},
	{
		icon: <BlogAvatar name="thetakeout" size={18} />,
		id: 3,
		name: 'The Takeout'
	},
	{
		icon: <BlogAvatar name="splinter" size={18} />,
		id: 2,
		name: 'Splinter'
	}
];


export const lastSharedBlogs = [
	{
		blogGroup: 'kotaku',
		createTimeMillis: 1552045361570,
		displayName: 'Cosplay',
		id: '1538697436',
		repostTimeMillis: 1553045361570
	},
	{
		blogGroup: 'gizmodo',
		displayName: 'Earther',
		createTimeMillis: 1551045261570,
		id: '1636390014',
		repostTimeMillis: 1553045261570
	},
	{
		blogGroup: 'splinter',
		displayName: 'Splinter',
		createTimeMillis: 1551045261570,
		id: '1635895473',
		repostTimeMillis: 1551245261570
	}
];