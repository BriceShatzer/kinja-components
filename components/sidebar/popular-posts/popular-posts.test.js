import * as React from 'react';
import PopularPosts from './popular-posts';
import SidebarPost from 'kinja-magma/models/SidebarPost';
import fixtures from '../__fixtures__/sidebar-posts';

const posts = fixtures.map(SidebarPost.fromJSON);
import { shallow } from 'enzyme';

describe('<PopularPosts />', () => {
	it('should render by default', () => {
		const result = shallow(
			<PopularPosts
				posts={posts}
			/>
		).dive();
		expect(result).toMatchSnapshot();
	});
	describe('Headline', () => {
		it('Recent from one author', () => {
			const result = shallow(
				<PopularPosts
					posts={posts.slice(0, 1)}
					isRecent
				/>
			).dive().find('popular-posts__HeaderText');
			expect(result).toMatchSnapshot();
		});
		it('Recent from two authors', () => {
			const result = shallow(
				<PopularPosts
					posts={posts.slice(0, 2)}
					isRecent
				/>
			).dive().find('popular-posts__HeaderText');
			expect(result).toMatchSnapshot();
		});
		it('Recent from more authors', () => {
			const result = shallow(
				<PopularPosts
					posts={posts.slice(0, 3)}
					isRecent
				/>
			).dive().find('popular-posts__HeaderText');
			expect(result).toMatchSnapshot();
		});
		it('Popular videos from blog', () => {
			const result = shallow(
				<PopularPosts
					posts={posts}
					currentBlog={{
						displayName: 'Blog Display Name',
						isSatire: true
					}}
				/>
			).dive().find('popular-posts__HeaderText');
			expect(result).toMatchSnapshot();
		});
		it('Popular from blog', () => {
			const result = shallow(
				<PopularPosts
					posts={posts}
					currentBlog={{
						displayName: 'Blog Display Name'
					}}
				/>
			).dive().find('popular-posts__HeaderText');
			expect(result).toMatchSnapshot();
		});
	});
});
