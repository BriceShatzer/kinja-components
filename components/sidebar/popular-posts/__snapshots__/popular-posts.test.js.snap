// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopularPosts /> Headline Popular from blog 1`] = `
<popular-posts__HeaderText
  dangerouslySetInnerHTML={
    Object {
      "__html": "<strong>Popular</strong> from Blog Display Name",
    }
  }
/>
`;

exports[`<PopularPosts /> Headline Popular videos from blog 1`] = `
<popular-posts__HeaderText
  dangerouslySetInnerHTML={
    Object {
      "__html": "<strong>Popular</strong> from Blog Display Name",
    }
  }
/>
`;

exports[`<PopularPosts /> Headline Recent from more authors 1`] = `
<popular-posts__HeaderText
  dangerouslySetInnerHTML={
    Object {
      "__html": "<strong>Recent</strong> from these authors",
    }
  }
/>
`;

exports[`<PopularPosts /> Headline Recent from one author 1`] = `
<popular-posts__HeaderText
  dangerouslySetInnerHTML={
    Object {
      "__html": "<strong>Recent</strong> from <a href=https://kinja.com/mborlay>Sam Barsanti</a>",
    }
  }
/>
`;

exports[`<PopularPosts /> Headline Recent from two authors 1`] = `
<popular-posts__HeaderText
  dangerouslySetInnerHTML={
    Object {
      "__html": "<strong>Recent</strong> from <a href=https://kinja.com/mborlay>Sam Barsanti</a> and <a href=>Clayton Purdom</a>",
    }
  }
/>
`;

exports[`<PopularPosts /> should render by default 1`] = `
<ThemeProvider
  theme={[Function]}
>
  <popular-posts__PopularPostsWrapper>
    <popular-posts__Header>
      <popular-posts__HeaderText
        dangerouslySetInnerHTML={
          Object {
            "__html": "<strong>Popular</strong> in your network",
          }
        }
      />
    </popular-posts__Header>
    <PopularPostItem
      gaEvent={[Function]}
      index={0}
      key="https://www.avclub.com/it-turns-out-that-people-dont-really-want-to-spend-1-0-1822570336"
      post={
        SidebarPost {
          "amazonPrice": undefined,
          "amazonPromoCode": undefined,
          "authorId": "5876237249236221114",
          "authorNameOrByline": "Sam Barsanti",
          "authorScreenName": "mborlay",
          "defaultBlogAvatar": Object {
            "format": "png",
            "frozenFormat": "png",
            "id": "no63bw902mddhwxtjtxh",
            "isAnimated": false,
          },
          "defaultBlogDisplayName": "The A.V. Club",
          "defaultBlogGroup": "avclub",
          "defaultBlogHost": "www.avclub.com",
          "defaultBlogId": "1636027099",
          "defaultBlogLogo": null,
          "defaultBlogRecircGroup": undefined,
          "headline": "It turns out that people don&#39;t really want to spend $1,000 on an iPhone<em></em>",
          "id": "1822570336",
          "image": Object {
            "altText": undefined,
            "format": "jpg",
            "frozenFormat": "jpg",
            "height": 0,
            "id": "tmc86rcc0hqcrztueilp",
            "originalHeight": undefined,
            "originalWidth": undefined,
            "uid": "tmc86rcc0hqcrztueilp",
            "width": 0,
          },
          "isVideo": undefined,
          "likeCount": 3,
          "originalProps": Object {
            "amazonPrice": undefined,
            "amazonPromoCode": undefined,
            "authorId": "5876237249236221114",
            "authorNameOrByline": "Sam Barsanti",
            "authorScreenName": "mborlay",
            "defaultBlogAvatar": Object {
              "format": "png",
              "frozenFormat": "png",
              "id": "no63bw902mddhwxtjtxh",
              "isAnimated": false,
            },
            "defaultBlogDisplayName": "The A.V. Club",
            "defaultBlogGroup": "avclub",
            "defaultBlogHost": "www.avclub.com",
            "defaultBlogId": "1636027099",
            "defaultBlogLogo": null,
            "defaultBlogRecircGroup": undefined,
            "headline": "It turns out that people don&#39;t really want to spend $1,000 on an iPhone<em></em>",
            "id": "1822570336",
            "image": Object {
              "altText": undefined,
              "format": "jpg",
              "frozenFormat": "jpg",
              "height": 0,
              "id": "tmc86rcc0hqcrztueilp",
              "originalHeight": undefined,
              "originalWidth": undefined,
              "uid": "tmc86rcc0hqcrztueilp",
              "width": 0,
            },
            "isVideo": undefined,
            "likeCount": 3,
            "permalink": "https://www.avclub.com/it-turns-out-that-people-dont-really-want-to-spend-1-0-1822570336",
            "plaintext": undefined,
            "publishTimeMillis": 1517355780336,
            "replyCount": 99,
            "score": 0,
            "viewCount": 200,
          },
          "permalink": "https://www.avclub.com/it-turns-out-that-people-dont-really-want-to-spend-1-0-1822570336",
          "plaintext": undefined,
          "publishTimeMillis": 1517355780336,
          "replyCount": 99,
          "score": 0,
          "viewCount": 200,
        }
      }
    />
    <PopularPostItem
      gaEvent={[Function]}
      index={1}
      key="https://www.avclub.com/deep-learning-technology-is-now-being-used-to-put-nic-c-1822514573"
      post={
        SidebarPost {
          "amazonPrice": undefined,
          "amazonPromoCode": undefined,
          "authorId": null,
          "authorNameOrByline": "Clayton Purdom",
          "authorScreenName": undefined,
          "defaultBlogAvatar": Object {
            "format": "png",
            "frozenFormat": "png",
            "id": "no63bw902mddhwxtjtxh",
            "isAnimated": false,
          },
          "defaultBlogDisplayName": "The A.V. Club",
          "defaultBlogGroup": "avclub",
          "defaultBlogHost": "www.avclub.com",
          "defaultBlogId": "1636027099",
          "defaultBlogLogo": null,
          "defaultBlogRecircGroup": undefined,
          "headline": "Deep learning technology is now being used to put Nic Cage in every movie",
          "id": "1822514573",
          "image": Object {
            "altText": undefined,
            "format": "png",
            "frozenFormat": "png",
            "height": 0,
            "id": "mrahpb6hsh07vuptempx",
            "originalHeight": undefined,
            "originalWidth": undefined,
            "uid": "mrahpb6hsh07vuptempx",
            "width": 0,
          },
          "isVideo": undefined,
          "likeCount": 5,
          "originalProps": Object {
            "amazonPrice": undefined,
            "amazonPromoCode": undefined,
            "authorId": null,
            "authorNameOrByline": "Clayton Purdom",
            "authorScreenName": undefined,
            "defaultBlogAvatar": Object {
              "format": "png",
              "frozenFormat": "png",
              "id": "no63bw902mddhwxtjtxh",
              "isAnimated": false,
            },
            "defaultBlogDisplayName": "The A.V. Club",
            "defaultBlogGroup": "avclub",
            "defaultBlogHost": "www.avclub.com",
            "defaultBlogId": "1636027099",
            "defaultBlogLogo": null,
            "defaultBlogRecircGroup": undefined,
            "headline": "Deep learning technology is now being used to put Nic Cage in every movie",
            "id": "1822514573",
            "image": Object {
              "altText": undefined,
              "format": "png",
              "frozenFormat": "png",
              "height": 0,
              "id": "mrahpb6hsh07vuptempx",
              "originalHeight": undefined,
              "originalWidth": undefined,
              "uid": "mrahpb6hsh07vuptempx",
              "width": 0,
            },
            "isVideo": undefined,
            "likeCount": 5,
            "permalink": "https://www.avclub.com/deep-learning-technology-is-now-being-used-to-put-nic-c-1822514573",
            "plaintext": undefined,
            "publishTimeMillis": 1517248980573,
            "replyCount": 69,
            "score": 0,
            "viewCount": 5,
          },
          "permalink": "https://www.avclub.com/deep-learning-technology-is-now-being-used-to-put-nic-c-1822514573",
          "plaintext": undefined,
          "publishTimeMillis": 1517248980573,
          "replyCount": 69,
          "score": 0,
          "viewCount": 5,
        }
      }
    />
    <PopularPostItem
      gaEvent={[Function]}
      index={2}
      key="https://www.avclub.com/in-this-week-s-savage-love-lesborama-1822515763"
      post={
        SidebarPost {
          "amazonPrice": undefined,
          "amazonPromoCode": undefined,
          "authorId": null,
          "authorNameOrByline": "Dan Savage",
          "authorScreenName": undefined,
          "defaultBlogAvatar": Object {
            "format": "png",
            "frozenFormat": "png",
            "id": "no63bw902mddhwxtjtxh",
            "isAnimated": false,
          },
          "defaultBlogDisplayName": "The A.V. Club",
          "defaultBlogGroup": "avclub",
          "defaultBlogHost": "www.avclub.com",
          "defaultBlogId": "1636027099",
          "defaultBlogLogo": null,
          "defaultBlogRecircGroup": undefined,
          "headline": "In this week’s Savage Love: Lesborama",
          "id": "1822515763",
          "image": null,
          "isVideo": undefined,
          "likeCount": 1,
          "originalProps": Object {
            "amazonPrice": undefined,
            "amazonPromoCode": undefined,
            "authorId": null,
            "authorNameOrByline": "Dan Savage",
            "authorScreenName": undefined,
            "defaultBlogAvatar": Object {
              "format": "png",
              "frozenFormat": "png",
              "id": "no63bw902mddhwxtjtxh",
              "isAnimated": false,
            },
            "defaultBlogDisplayName": "The A.V. Club",
            "defaultBlogGroup": "avclub",
            "defaultBlogHost": "www.avclub.com",
            "defaultBlogId": "1636027099",
            "defaultBlogLogo": null,
            "defaultBlogRecircGroup": undefined,
            "headline": "In this week’s Savage Love: Lesborama",
            "id": "1822515763",
            "image": null,
            "isVideo": undefined,
            "likeCount": 1,
            "permalink": "https://www.avclub.com/in-this-week-s-savage-love-lesborama-1822515763",
            "plaintext": undefined,
            "publishTimeMillis": 1517385600763,
            "replyCount": 2,
            "score": 0,
            "viewCount": undefined,
          },
          "permalink": "https://www.avclub.com/in-this-week-s-savage-love-lesborama-1822515763",
          "plaintext": undefined,
          "publishTimeMillis": 1517385600763,
          "replyCount": 2,
          "score": 0,
          "viewCount": undefined,
        }
      }
    />
  </popular-posts__PopularPostsWrapper>
</ThemeProvider>
`;
